src/UdtSocket.cs

```
@@ -21,6 +21,20 @@ public UdtSocket(AddressFamily addressFamily, SocketType socketType)
             }
         }
 
+        public int Bind(Socket socket)
+        {
+            try
+            {
+                int status = UDT.s_UDTUnited.bind(mSocketId, socket);
+                mLocalEndPoint = socket.LocalEndPoint as IPEndPoint;

```

using the `as` dynamic cast should only be done when you're going to check for null afterwards; but if you don't want to check (because the type cast must work, like an assertion style) then you should use a static cast:

```
mLocalEndPoint = (IPEndPoint)socket.LocalEndPoint;
```

This way, if the cast fails, you get a clearer `InvalidCastException`, instead of the weird `NullReferenceException` that you would get if you use `as`

---

there's another use of `as` in this PR which isn't solved yet